### PR TITLE
add the wrapper.getComponent method and corresponding documents.

### LINKED ISSUE #1711: finding components with `find` is deprecated and will be removed in the next major version. Use `findComponent` instead.
When I was executing my unit tests, which used the wrapper.get() method, it outputed the following content:

image

### PR DESCRIPTION
**Pull request by TinyWisp**
Closes #1711 

feat: add the wrapper.getComponent method
docs: create the /docs/api/wrapper/getComponent.md, and add the deprecation message to the /docs/api/wrapper/get.md
docs(zh): create the /docs/zh/api/wrapper/getComponent.md, and add the deprecation message to the /docs/zh/api/wrapper/get.md

### CODE REVIEW

File: packages/test-utils/src/wrapper.js [hunk 1]
```diff
@@ -237,6 +237,20 @@ export default class Wrapper implements BaseWrapper {
...
+  /**
+   * Gets first node in tree of the current wrapper that
+   * matches the provided selector.
+   */
+  getComponent(rawSelector: Selector): Wrapper {
+    this.__warnIfDestroyed()
+
+    const found = this.findComponent(rawSelector)
+    if (found instanceof ErrorWrapper) {
+      throw new Error(`Unable to find ${rawSelector} within: ${this.html()}`)
```
Comment (afontcu) [inline packages/test-utils/src/wrapper.js]:
```suggestion
      throw new Error(`Unable to get ${rawSelector} within: ${this.html()}`)
```

File: docs/api/wrapper/getComponent.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import Foo from './Foo.vue'
+import Bar from './Bar.vue'
+
+const wrapper = mount(Foo)
+
+// similar to `wrapper.findComponent`.
+// `getComponent` will throw an error if an element is not found. `findComponent` will do nothing.
+expect(wrapper.getComponent(Bar)) // => finds Bar by component instance
+expect(wrapper.getComponent({ name: 'bar' })) // => finds Bar by `name`
+expect(wrapper.getComponent({ ref: 'bar' })) // => finds Bar by `ref`
```
Comment (afontcu) [inline docs/api/wrapper/getComponent.md]:
```suggestion
expect(wrapper.getComponent(Bar)) // => gets Bar by component instance
expect(wrapper.getComponent({ name: 'bar' })) // => gets Bar by `name`
expect(wrapper.getComponent({ ref: 'bar' })) // => gets Bar by `ref`
```

File: docs/zh/api/wrapper/getComponent.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import Foo from './Foo.vue'
+import Bar from './Bar.vue'
+
+const wrapper = mount(Foo)
+
+// 和 `wrapper.findComponent` 相似。
+// 如果 `getComponent` 没有找到任何元素将会抛出一个而错误。`findComponent` 则不会做任何事。
+expect(wrapper.getComponent(Bar)) // => finds Bar by component instance
+expect(wrapper.getComponent({ name: 'bar' })) // => finds Bar by `name`
+expect(wrapper.getComponent({ ref: 'bar' })) // => finds Bar by `ref`
```
Comment (afontcu) [inline docs/zh/api/wrapper/getComponent.md]:
```suggestion
expect(wrapper.getComponent(Bar)) // => gets Bar by component instance
expect(wrapper.getComponent({ name: 'bar' })) // => gets Bar by `name`
expect(wrapper.getComponent({ ref: 'bar' })) // => gets Bar by `ref`
```

File: docs/zh/api/wrapper/getComponent.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+expect(() => wrapper.getComponent({ name: 'does-not-exist' }))
+  .to.throw()
+  .with.property(
+    'message',
+    "Unable to find a component named 'does-not-exist' within: <div>the actual DOM here...</div>"
```
Comment (afontcu) [inline docs/zh/api/wrapper/getComponent.md]:
```suggestion
    "Unable to get a component named 'does-not-exist' within: <div>the actual DOM here...</div>"
```

File: docs/api/wrapper/getComponent.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+expect(() => wrapper.getComponent({ name: 'does-not-exist' }))
+  .to.throw()
+  .with.property(
+    'message',
+    "Unable to find a component named 'does-not-exist' within: <div>the actual DOM here...</div>"
```
Comment (afontcu) [inline docs/api/wrapper/getComponent.md]:
```suggestion
    "Unable to get a component named 'does-not-exist' within: <div>the actual DOM here...</div>"
```

**Review by afontcu — COMMENTED** [review]
Looks good, just suggested a small tweak (in several files, but a single tweak after all!) that bitten us in the past :)

**Review by lmiller1990 — APPROVED** [review]
Just need to make the recommended changes, and I will release this. Thanks!

**Pull request merged**